docs/development/extensions-contrib/k8s-jobs.md

```diff
@@ -115,33 +75,197 @@ You can keep your Dockerfile the same but you must have a sidecar spec like so:
            - arg2 
 ```
 
-The following roles must also be accessible. An example spec could be: 
+For both of these adapters, you can add optional labels to your K8s jobs / pods if you need them by using the following configuration:
+`druid.indexer.runner.labels: '{"key":"value"}'`
+Annotations are the same with:
+`druid.indexer.runner.annotations: '{"key":"value"}'`
+
+All other configurations you had for the middle manager tasks must be moved under the overlord with one caveat, you must specify javaOpts as an array:
+`druid.indexer.runner.javaOptsArray`, `druid.indexer.runner.javaOpts` is no longer supported.
+
+If you are running without a middle manager you need to also use `druid.processing.intermediaryData.storage.type=deepstore`
+
+### Custom Template Pod Adapter
+The custom template pod adapter allows you to specify a pod template file per task type for more flexibility on how to define your pods. This adapter expects a [Pod Template](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates) to be available on the overlord's file system. This pod template is used as the base of the pod spec for the Kubernetes Job. You can override things like labels, environment variables, resources, annotation, or even the base image with this template. To enable this pod adapter you can specify the runtime property `druid.indexer.runner.k8s.adapter.type: customTemplateAdapter`
+
+The base pod template must be specified as the runtime property `druid.indexer.runner.k8s.podTemplate.base: /path/to/basePodSpec.yaml`
+
+Task specific pod templates must be specified as the runtime property `druid.indexer.runner.k8s.podTemplate.{taskType}: /path/to/taskSpecificPodSpec.yaml` where {taskType} is the name of the task type i.e `index_parallel`
+
+The following is an example Pod Template that uses the regular druid docker image.
+```
+apiVersion: "v1"
+kind: "PodTemplate"
+template:
+  metadata:
+    annotations:
+      custom-annotation: "hello"
+    labels:
+      custom-label: "hello"
+  spec:
+    affinity: {}
+    containers:
+    - command:
+        - sh
+        - -c
+        - |
+          /peon.sh /druid/data 1
+      env:
+      - name: CUSTOM_ENV_VARIABLE
+        value: "hello"
+      image: apache/druid:26.0.0
+      name: main
+      ports:
+      - containerPort: 8091
+        name: druid-tls-port
+        protocol: TCP
+      - containerPort: 8100
+        name: druid-port
+        protocol: TCP
+      resources:
+        limits:
+          cpu: "1"
```

Comment:
Do we need to set a limit?

Reply:
that depends on what you are doing but usually it's best practice to do so

Reply:
https://home.robusta.dev/blog/stop-using-cpu-limits - I read this blog post that seems to advise against setting CPU limits

Reply:
i would say my intention with this template is to show all the possible configurations you can set on the pod template, it's easier to see the configuration in docs and look up what it does and remove it if not needed than to try to figure out if a property you found somewhere else can be set

it's not necessarily intended to be a recommended configuration